src/common/config_parser.cpp

```diff
@@ -861,6 +875,21 @@ Ptr<Options> ConfigParser::parseOptions(int argc, char** argv, bool doValidate){
     cli::processPaths(config_, cli::InterpolateEnvVars, PATHS);
   }
 
+  // Option shortcuts for input from STDIN for trainer and scorer
+  if(mode_ == cli::mode::training || mode_ == cli::mode::scoring) {
+    auto trainSets = get<std::vector<std::string>>("train-sets");
+    YAML::Node config;
+    // Assume the input will come from STDIN if --tsv is set but no --train-sets are given
+    if(get<bool>("tsv") && trainSets.empty()) {
+      config["train-sets"].push_back("stdin");
+    // Assume the input is in TSV format if --train-sets is set to "stdin"
+    } else if(trainSets.size() == 1 && trainSets[0] == "stdin") {
+      config["tsv"] = true;
```

Comment:
It works, and I would say it's correct. My idea was that for simplicity and to avoid potential special cases, it might be better to set TSV always if the training is from stdin, regardless of the number of streams.

I also added a regression test for training with `cat file | marian --type lm -t stdin`.

Reply:
BTW, at the moment marian does not know about the correct number of streams for different model types and does not put restrictions on the number of streams. The only validation is len(train-set) == len(vocabs). It's possible to train a LM with parallel data, while training a seq2seq model with a single train-set and vocab results in a segmentation fault. This probably deserves an issue on GitHub.